Cpp_test/lib/ipc-shm.cpp

```diff
@@ -0,0 +1,113 @@
+#include "ipc-shm.h"
+
+IPCShm::~IPCShm()
+{
+
+	if (pthread_condattr_destroy(&(this->cond_attr)) != 0)
+	{
+		perror("Error at pthread_condattr_destroy()\n");
+	}
+	
+	if (pthread_mutexattr_destroy(&(this->mutex_attr)) != 0)
+	{
+		perror("Error at pthread_mutexattr_destroy()\n");
+	}
+	if (this->shm_ptr != nullptr)
+	{
+		munmap(this->shm_ptr, this->shm_size_in_bytes);
+	}
+}
+
+void IPCShm::init_cond()
+{
+
+	if (pthread_condattr_init(&(this->cond_attr)) != 0)
+	{
+		throw std::runtime_error("Error at pthread_condattr_init()");
+	}
+
+	if (pthread_condattr_setpshared(&(this->cond_attr), PTHREAD_PROCESS_SHARED) != 0)
+	{
+		throw std::runtime_error("Error at pthread_condattr_setpshared()");
+	}
+
+	/* Destory the cond attributes object */
+	if (pthread_condattr_destroy(&(this->cond_attr)) != 0)
```

Comment:
Why destroying the condattr just after initialized it?

Reply:
I misunderstood in the first place, destroy function is not placed here, it should be in the cleanup place before exit.